core/src/main/scala/kafka/tools/ConsoleProducer.scala

```diff
@@ -302,22 +304,33 @@ object ConsoleProducer {
       if (props.containsKey("key.separator"))
         keySeparator = props.getProperty("key.separator")
       if (props.containsKey("parse.headers"))
-        parseHeader = props.getProperty("parse.headers").trim.equalsIgnoreCase("true")
+        parseHeaders = props.getProperty("parse.headers").trim.equalsIgnoreCase("true")
       if (props.containsKey("headers.delimiter"))
         headersDelimiter = props.getProperty("headers.delimiter")
       if (props.containsKey("headers.separator"))
         headersSeparator = props.getProperty("headers.separator")
-        headerSeparatorPattern = Pattern.compile(headersSeparator)
+        headersSeparatorPattern = Pattern.compile(headersSeparator)
       if (props.containsKey("headers.key.separator"))
-        headerKeySeparator = props.getProperty("headers.key.separator")
+        headersKeySeparator = props.getProperty("headers.key.separator")
       if (props.containsKey("ignore.error"))
         ignoreError = props.getProperty("ignore.error").trim.equalsIgnoreCase("true")
       if (headersDelimiter == headersSeparator)
         throw new KafkaException("headers.delimiter and headers.separator may not be equal")
-      if (headersDelimiter == headerKeySeparator)
+      if (headersDelimiter == headersKeySeparator)
         throw new KafkaException("headers.delimiter and headers.key.separator may not be equal")
-      if (headersSeparator == headerKeySeparator)
+      if (headersSeparator == headersKeySeparator)
         throw new KafkaException("headers.separator and headers.key.separator may not be equal")
+      if (props.containsKey("null.marker")) {
+        nullMarker = props.getProperty("null.marker")
+      }
```

Comment:
nit: I would remove the curly braces here to stay inline with the style of the other `if`s.